pkg/clusteroperator/clusteroperators.go

```diff
@@ -0,0 +1,273 @@
+		glog.V(100).Infof("Error to list ClusterOperators")
+
+		return err
+	}
+
+	if len(coList.Items) < 1 {
+		glog.V(100).Infof("Cluster doesn't have ClusterOperator deployed, bad cluster state ...")
+
+		return fmt.Errorf("ClusterOperator list is empty")
+	}
+
+	builder.ObjectList = coList
+
+	for _, co := range coList.Items {
+		glog.V(100).Infof("builder Discover() ClusterOperator contents: %v", co.ObjectMeta.Name)
+	}
+
+	return err
+}
+
+// WaitForAllAvailable waits for all ClusterConfigs to be available for a time duration up to the timeout.
+func (builder *CoListBuilder) WaitForAllAvailable(availableDuration time.Duration,
+	timeout time.Duration) (bool, error) {
+	glog.V(100).Infof("Wait for each ClusterOperator from the ClusterOperatorsList reaches "+
+		"Available state up to duration of %v", timeout)
+
+	isAllCoAvailable := true
+
+	// Wait 5 secs in each iteration before condition function () returns true or errors or times out
+	// after availableDuration
+	err := wait.PollImmediate(fiveScds, timeout, func() (bool, error) {
+
+		isAllCoAvailable = true
+
+		// check if clusteroperator is available every 5 seconds during entire availableDuration time period
+		_ = wait.PollImmediate(fiveScds, availableDuration, func() (done bool, err error) {
+
+			err = builder.Discover()
+
+			if err != nil {
+				return false, err
+			}
+
+			// iterate through the ClusterOperators in the list.
+			for _, clusteroperator := range builder.ObjectList.Items {
+				for _, condition := range clusteroperator.Status.Conditions {
+					if condition.Type == "Available" && condition.Status != isTrue {
+						isAllCoAvailable = false
+					}
+
+					glog.V(100).Infof("ClusterOperators: %v not available",
+						clusteroperator.Name)
+				}
+			}
+
+			// Here we are always returning "false, nil" so we keep iterating throughout the stableInterval
+			// of the inner wait.pollImmediate loop, until we time out.
+			return false, nil
+		})
+
+		if isAllCoAvailable {
+			glog.V(100).Infof("All ClusterOperators were found available during availableDuration: %v",
+				availableDuration)
+
+			return true, nil
+		}
+
+		glog.V(100).Infof("Not all ClusterOperators were found available during availableDuration: %v, "+
+			"retrying ...", availableDuration)
+
+		// keep iterating in the outer wait.PollImmediate waiting for cluster to be stable.
+		return false, nil
+
+	})
+
+	if err == nil {
+		glog.V(100).Infof("All ClusterOperators were found available  during availableDuration: %v",
+			availableDuration)
+	} else {
+		// Here err is "timed out waiting for the condition"
+		glog.V(100).Infof("Not all ClusterOperators were found available  during availableDuration: %v",
+			availableDuration)
+	}
+
+	return false, err
+}
+
+// WaitForAllStopProgressing waits until all clusteroperators will finish with a progressing (PROGRESSING=False).
+func (builder *CoListBuilder) WaitForAllStopProgressing(availableDuration time.Duration, timeout time.Duration) error {
```

Comment:
Please can you re-name this function as well to WaitToBeAllStopProgressingFor

Reply:
done